ui/job-view/redux/stores/pushes.js

```diff
@@ -0,0 +1,385 @@
+import pick from 'lodash/pick';
+import keyBy from 'lodash/keyBy';
+import max from 'lodash/max';
+
+import { parseQueryParams } from '../../../helpers/url';
+import {
+  getAllUrlParams,
+  getQueryString,
+  getUrlParam,
+  replaceLocation,
+} from '../../../helpers/location';
+import PushModel from '../../../models/push';
+import { isUnclassifiedFailure } from '../../../helpers/job';
+import FilterModel from '../../../models/filter';
+import JobModel from '../../../models/job';
+import { thEvents } from '../../../helpers/constants';
+
+import { notify } from './notifications';
+import { setSelectedJob, clearSelectedJob } from './selectedJob';
+
+export const LOADING = 'LOADING';
+export const ADD_PUSHES = 'ADD_PUSHES';
+export const CLEAR_PUSHES = 'CLEAR_PUSHES';
+export const SET_PUSHES = 'SET_PUSHES';
+export const RECALCULATE_UNCLASSIFIED_COUNTS =
+  'RECALCULATE_UNCLASSIFIED_COUNTS';
+export const UPDATE_JOB_MAP = 'UPDATE_JOB_MAP';
+
+const DEFAULT_PUSH_COUNT = 10;
+// Keys that, if present on the url, must be passed into the push
+// polling endpoint
+const PUSH_POLLING_KEYS = ['tochange', 'enddate', 'revision', 'author'];
+const PUSH_FETCH_KEYS = [...PUSH_POLLING_KEYS, 'fromchange', 'startdate'];
+
+const getRevisionTips = pushList => {
+  return {
+    revisionTips: pushList.map(push => ({
+      revision: push.revision,
+      author: push.author,
+      title: push.revisions[0].comments.split('\n')[0],
+    })),
+  };
+};
+
+const getLastModifiedJobTime = jobMap => {
+  const latest =
+    max(Object.values(jobMap).map(job => new Date(`${job.last_modified}Z`))) ||
+    new Date();
+
+  latest.setSeconds(latest.getSeconds() - 3);
+  return latest;
+};
+
+/**
+ * Loops through the map of unclassified failures and checks if it is
+ * within the enabled tiers and if the job should be shown. This essentially
+ * gives us the difference in unclassified failures and, of those jobs, the
+ * ones that have been filtered out
+ */
+const doRecalculateUnclassifiedCounts = jobMap => {
+  const filterModel = new FilterModel();
+  const tiers = filterModel.urlParams.tier;
+  let allUnclassifiedFailureCount = 0;
+  let filteredUnclassifiedFailureCount = 0;
+
+  Object.values(jobMap).forEach(job => {
+    if (isUnclassifiedFailure(job)) {
+      if (tiers.includes(String(job.tier))) {
+        // if (filterModel.showJob(job)) {
```

Comment:
delete line

Reply:
Can some of these if statements be combined instead of nesting them? :)

Reply:
Yep.  Oops.  :)